tsl/src/compression/compression.c

```diff
@@ -2296,18 +2297,21 @@ build_index_scankeys_using_slot(Oid hypertable_relid, Relation in_rel, Relation
 			AttrNumber attnum = AttrOffsetGetAttrNumber(i);
 			char *attname = get_attname(RelationGetRelid(index_rel), attnum, false);
 
-			/* If we are at the last attribute, check its the sequence number attribute.
-			 * This means we found all other attributes on the hypertable and this could be our
-			 * index.
-			 */
-			if (index_rel->rd_index->indnatts - 1 == i)
+			/* Make sure we find columns in key columns in order to select the right index */
+			bool idx_col_found = false;
+			while ((i = bms_next_member(key_columns, i)) > 0)
```

Comment:
When you have a `Relation`, might be better to use `attnumAttName`, because it doesn't do a catalog lookup again, but looks into the `Relation`. Also there is the counterpart function `attnameAttNum`, so you might be able to replace this entire loop with something like `bms_is_member(attnameAttNum(table_rel, attnumAttName(index_rel)))`.

Reply:
Don't have the out_rel opened but works for the index.